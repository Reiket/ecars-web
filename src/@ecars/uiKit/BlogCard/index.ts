import {BlogCardContent} from '@ecars/uiKit/BlogCard/components/BlogCardContent';
import {BlogCardComponent} from '@ecars/uiKit/BlogCard/BlogCardComponent';
import {BlogCardImage} from '@ecars/uiKit/BlogCard/components/BlogCardImage';

export const BlogCard = Object.assign(BlogCardComponent, {
  Content: BlogCardContent,
  Image: BlogCardImage,
});
