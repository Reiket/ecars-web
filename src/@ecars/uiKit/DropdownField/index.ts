import {DropdownFieldComponent} from '@ecars/uiKit/DropdownField/DropdownFieldComponent';
import {DropdownFieldWrapper} from '@ecars/uiKit/DropdownField/components/DropdownFieldWrapper';

export const DropdownField = Object.assign(DropdownFieldComponent, {
  Wrapper: DropdownFieldWrapper,
});
