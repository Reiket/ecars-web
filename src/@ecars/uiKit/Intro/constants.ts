import {withBlockClass} from 'ecars-web-lib';
import {PageWrapper} from '@ecars/uiKit/Page/components/PageWrapper';

export const IntroWrapperHOC = withBlockClass(PageWrapper, 'intro-block');
export const BenefitsWrapperHOC = withBlockClass(PageWrapper, 'benefits');

export interface BenefitItemType {
  icon: string;
  title: string;
  text: string;
}

export const benefitsItemConfig: BenefitItemType[] = [
  {
    icon: '/assets/images/benefits/01.svg',
    title: 'Professional approach to clients',
    text: 'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.',
  },
  {
    icon: '/assets/images/benefits/02.svg',
    title: 'Protect all payments',
    text: 'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.',
  },
  {
    icon: '/assets/images/benefits/01.svg',
    title: 'Real reviews from clients',
    text: 'Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.',
  },
];
