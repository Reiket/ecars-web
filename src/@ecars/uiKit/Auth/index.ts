import {AuthLayout} from '@ecars/uiKit/Auth/components/AuthLayout';
import {AuthPageWrapperHOC} from '@ecars/uiKit/Auth/constants';

export const AuthPage = Object.assign(AuthLayout, {
  Wrapper: AuthPageWrapperHOC,
});