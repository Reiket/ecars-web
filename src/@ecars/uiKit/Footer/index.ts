import {FooterComponent} from '@ecars/uiKit/Footer/FooterComponent';
import {FooterCopyWrapperHOC, FooterWrapperHOC} from '@ecars/uiKit/Footer/constants';
import {FooterCopy} from '@ecars/uiKit/Footer/components/FooterCopy';
import {FooterCopyList} from '@ecars/uiKit/Footer/components/FooterCopyList';
import {FooterColumn} from '@ecars/uiKit/Footer/components/FooterColumn';
import {FooterLogoBlock} from '@ecars/uiKit/Footer/components/FooterLogoBlock';
import {FooterLinkList} from '@ecars/uiKit/Footer/components/FooterLinkList';

export const Footer = Object.assign(FooterComponent, {
  Wrapper: FooterWrapperHOC,
  Column: FooterColumn,
  LogoBlock: FooterLogoBlock,
  LinkList: FooterLinkList,
});

export const Copy = Object.assign(FooterCopy, {
  Wrapper: FooterCopyWrapperHOC,
  List: FooterCopyList,
});
