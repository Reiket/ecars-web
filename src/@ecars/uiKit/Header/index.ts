import {HeaderComponent} from '@ecars/uiKit/Header/HeaderComponent';
import {HeaderNavbarWrapperHOC, HeaderWrapperHOC} from '@ecars/uiKit/Header/constants';
import {HeaderNavbar} from '@ecars/uiKit/Header/components/HeaderNavbar';
import {HeaderUtils} from '@ecars/uiKit/Header/components/HeaderUtils';
import {HeaderBurger} from '@ecars/uiKit/Header/components/HeaderBurger';
import {HeaderNavbarList} from '@ecars/uiKit/Header/components/HeaderNavbarList';

export const Header = Object.assign(HeaderComponent, {
  Wrapper: HeaderWrapperHOC,
  Utils: HeaderUtils,
  Burger: HeaderBurger,
});

export const Navbar = Object.assign(HeaderNavbar, {
  Wrapper: HeaderNavbarWrapperHOC,
  List: HeaderNavbarList,
});
