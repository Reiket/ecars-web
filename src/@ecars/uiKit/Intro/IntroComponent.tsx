import type {FC} from 'react';

import {Benefits, Intro} from '@ecars/uiKit/Intro';
import type {ElementProps} from 'ecars-web-lib';
import {cn} from 'ecars-web-lib';

export const IntroComponent: FC<ElementProps> = ({block}) => (
  <div className={cn(block, 'intro')}>
    <Intro.Block />
    <Benefits />
  </div>
);
