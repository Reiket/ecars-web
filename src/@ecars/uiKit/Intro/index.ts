import {IntroBlock} from '@ecars/uiKit/Intro/components/IntroBlock';
import {IntroWrapperHOC, BenefitsWrapperHOC} from '@ecars/uiKit/Intro/constants';
import {IntroBenefits} from '@ecars/uiKit/Intro/components/introBenefits';
import {IntroComponent} from '@ecars/uiKit/Intro/IntroComponent';
import {IntroBenefitItem} from '@ecars/uiKit/Intro/components/IntroBenefitsItem';

export const Intro = Object.assign(IntroComponent, {
  Wrapper: IntroWrapperHOC,
  Block: IntroBlock,
});

export const Benefits = Object.assign(IntroBenefits, {
  Wrapper: BenefitsWrapperHOC,
  Item: IntroBenefitItem,
});
