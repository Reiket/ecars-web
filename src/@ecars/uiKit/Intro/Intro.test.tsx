import {render} from '@testing-library/react';
import {Intro} from '@ecars/uiKit/Intro';

describe('Intro Component', () => {
  test('render component correctly', () => {
    const {container} = render(<Intro />);
    expect(container).toMatchSnapshot();
    expect(container).toBeInTheDocument();
  });
});
