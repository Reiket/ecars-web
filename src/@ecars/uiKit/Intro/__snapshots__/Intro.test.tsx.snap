// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Intro Component > render component correctly 1`] = `
<div>
  <div
    class="intro"
  >
    <div
      class="intro-block"
    >
      <section
        class="intro-block__section section"
      >
        <div
          class="intro-block__container container container--lg"
        >
          <div
            class="intro-block__body"
          >
            <h1
              class="intro-block__title page-title"
            >
              Car import services with delivery to your doorstep.
            </h1>
            <div
              class="intro-block__block"
            >
              <button
                class="intro-block__button button button--big button--green"
                data-testid="buttonTestId"
                type="button"
              >
                Get a quote
              </button>
              <ul
                class="intro-block__social social"
              >
                <li
                  class="social__item"
                >
                  <a
                    class="intro-block__link link link--dark link--icon"
                    data-testid="routerLinkTestId"
                    href="https://wa.me/971558119024"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <svg
                      data-testid="leftIconLinkTestId"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M260.062 32C138.605 32 40.134 129.701 40.134 250.232c0 41.23 11.532 79.79 31.559 112.687L32 480l121.764-38.682c31.508 17.285 67.745 27.146 106.298 27.146C381.535 468.464 480 370.749 480 250.232 480 129.701 381.535 32 260.062 32zm109.362 301.11c-5.174 12.827-28.574 24.533-38.899 25.072-10.314.547-10.608 7.994-66.84-16.434-56.225-24.434-90.052-83.844-92.719-87.67-2.669-3.812-21.78-31.047-20.749-58.455 1.038-27.413 16.047-40.346 21.404-45.725 5.351-5.387 11.486-6.352 15.232-6.413 4.428-.072 7.296-.132 10.573-.011 3.274.124 8.192-.685 12.45 10.639 4.256 11.323 14.443 39.153 15.746 41.989 1.302 2.839 2.108 6.126.102 9.771-2.012 3.653-3.042 5.935-5.961 9.083-2.935 3.148-6.174 7.042-8.792 9.449-2.92 2.665-5.97 5.572-2.9 11.269 3.068 5.693 13.653 24.356 29.779 39.736 20.725 19.771 38.598 26.329 44.098 29.317 5.515 3.004 8.806 2.67 12.226-.929 3.404-3.599 14.639-15.746 18.596-21.169 3.955-5.438 7.661-4.373 12.742-2.329 5.078 2.052 32.157 16.556 37.673 19.551 5.51 2.989 9.193 4.529 10.51 6.9 1.317 2.38.901 13.531-4.271 26.359z"
                      />
                    </svg>
                    +971 55 811 9024
                  </a>
                </li>
                <li
                  class="social__item"
                >
                  <a
                    class="intro-block__link link link--dark link--icon"
                    data-testid="routerLinkTestId"
                    href="mailto:request@example.com"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <svg
                      data-testid="leftIconLinkTestId"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                      />
                    </svg>
                    request@example.com
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </section>
    </div>
    <div
      class="benefits"
    >
      <section
        class="benefits__section section"
      >
        <div
          class="benefits__container container container--lg"
        >
          <div
            class="benefits__body"
          >
            <ul
              class="benefits__list"
            >
              <li
                class="benefits__item"
              >
                <div
                  class="benefits__icon"
                >
                  <img
                    alt="benefits_icon"
                    src="/assets/images/benefits/01.svg"
                  />
                </div>
                <div
                  class="benefits__info"
                >
                  <h4
                    class="benefits__title title-benefits-list"
                  >
                    Professional approach to clients
                  </h4>
                  <p
                    class="benefits__text"
                  >
                    Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
                  </p>
                </div>
              </li>
              <li
                class="benefits__item"
              >
                <div
                  class="benefits__icon"
                >
                  <img
                    alt="benefits_icon"
                    src="/assets/images/benefits/02.svg"
                  />
                </div>
                <div
                  class="benefits__info"
                >
                  <h4
                    class="benefits__title title-benefits-list"
                  >
                    Protect all payments
                  </h4>
                  <p
                    class="benefits__text"
                  >
                    Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
                  </p>
                </div>
              </li>
              <li
                class="benefits__item"
              >
                <div
                  class="benefits__icon"
                >
                  <img
                    alt="benefits_icon"
                    src="/assets/images/benefits/01.svg"
                  />
                </div>
                <div
                  class="benefits__info"
                >
                  <h4
                    class="benefits__title title-benefits-list"
                  >
                    Real reviews from clients
                  </h4>
                  <p
                    class="benefits__text"
                  >
                    Amet minim mollit non deserunt ullamco est sit aliqua dolor do amet sint.
                  </p>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
